components/formats-bsd/src/loci/formats/in/OBFReader.java

```diff
@@ -561,131 +651,353 @@ private String readString() throws IOException {
+    long seekDestination = 0;
+    long extraSkipBytes = stackByteOffset;
 
-        byte[] input = new byte[8192];
-        int end = isFLIM ? getSizeZ() - 1 : no;
+    if (stack.flushPoints != null && stack.flushBlockSize != 0) {
+      int flushBlockIndex = (int)(stackByteOffset / stack.flushBlockSize);
 
-        while (currentInflatedFrame.number != end) {
-          int offset = 0;
-          while (offset != bytes.length) {
-            if (inflater.needsInput()) {
-              final long remainder = stack.position + stack.length - in.getFilePointer();
-              if (remainder > 0) {
-                final int length = remainder > input.length ? input.length : (int) remainder;
+      if (flushBlockIndex > 0) {
+        seekDestination = stack.flushPoints.get(flushBlockIndex - 1);
+        extraSkipBytes -= flushBlockIndex * stack.flushBlockSize;
+      }
+    }
 
-                in.read(input, 0, length);
-                inflater.setInput(input, 0, length);
-              }
-              else if (isFLIM && remainder == 0 && currentInflatedFrame.number >= 0) {
-                offset = bytes.length;
-                continue;
-              }
-              else {
-                throw new FormatException("Corrupted zlib compression");
-              }
-            }
-            else if (inflater.needsDictionary()) {
-              throw new FormatException("Unsupported zlib compression");
-            }
-            try {
-              offset += inflater.inflate(bytes, offset, bytes.length - offset);
-            }
-            catch (DataFormatException exception) {
-              throw new FormatException(exception.getMessage());
-            }
-          }
-          ++ currentInflatedFrame.number;
-        }
+    if (stack.compression) {
+      state.inflater = new Inflater(seekDestination != 0);
+
+      if (state.inflateInputBuffer == null) {
+        state.inflateInputBuffer = new byte[8192];
       }
-      if (isFLIM) {
-        for (int yy=y; yy<y+h; yy++) {
-          for (int xx=x; xx<x+w; xx++) {
-            int src = getSizeZ() * bytesPerPixel * ((yy * getSizeX()) + xx) + no * bytesPerPixel;
-            int dest = ((yy - y) * w * bytesPerPixel) + ((xx - x) * bytesPerPixel);
-            System.arraycopy(bytes, src, buffer, dest, bytesPerPixel);
-          }
-        }
+    }
+
+    if (!hasChunks) {
+      in.seek(stack.position + seekDestination);
+      skipBytes(stack, extraSkipBytes);
+      return;
+    }
+
+    switchChunk(stack, Collections.binarySearch(
+      stack.chunkLogicalPositions, seekDestination));
+
+    in.seek(state.chunkFileStart +
+      seekDestination - state.chunkLogicalStart);
+
+    skipBytes(stack, extraSkipBytes);
+  }
+
+  private void skipBytes(Stack stack, long byteCount)
+    throws IOException, FormatException
+  {
+    boolean hasChunks = stack.chunkLogicalPositions != null;
+
+    if (!stack.compression && !hasChunks) {
+      in.seek(in.getFilePointer() + byteCount);
```

Comment:
This can be simplified to ```in.skipBytes(byteCount)```, since ```RandomAccessInputStream``` has a ```skipBytes(long)``` method.

Reply:
Thanks I change the corresponding code.